newIDE/app/src/MainFrame/EditorTabs/EditorTabsHandler.js

```diff
@@ -289,3 +289,24 @@ export const getEventsFunctionsExtensionEditor = (
 
   return null;
 };
+
+export const moveTabToPosition = (
+  editorTabsState: EditorTabsState,
+  fromIndex: number,
+  toIndex: number
+): EditorTabsState => {
+  if (toIndex < fromIndex) toIndex += 1;
+  const currentEditorTabs = [...getEditors(editorTabsState)];
+  const movingTab = currentEditorTabs[fromIndex];
+  currentEditorTabs.splice(fromIndex, 1);
+  currentEditorTabs.splice(toIndex, 0, movingTab);
```

Comment:
I wonder if the `toIndex` increment should be outside of this method because even if there is no bug, the method probably not do what it says it does.

If we do this step by step:
![TabDrop](https://user-images.githubusercontent.com/2611977/157723352-701521ee-2381-4d2c-b850-15b1d7d716d0.png)

* E from 4 to 2
```
0 1 2 3 4 5 6
A B C D E F G

  currentEditorTabs.splice(4, 1);

0 1 2 3 4 5 6
A B C D F G

  currentEditorTabs.splice(2 + 1, 0, movingTab);

0 1 2 3 4 5 6
A B C E D F G
```
It actually moved E from 4 to 3.

* C from 2 to 4

```
0 1 2 3 4 5 6
A B C D E F G

  currentEditorTabs.splice(2, 1);

0 1 2 3 4 5 6
A B D E F G

  currentEditorTabs.splice(4, 0, movingTab);

0 1 2 3 4 5 6
A B D E C F G
```

In the other case, I see no issue.